lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -515,68 +576,345 @@ def make_latest_pki_issuer_default(self, mount: str) -> None:
         except (TypeError, KeyError):
             logger.error("Issuers config is not yet created")
 
-    def _get_autounseal_policy_name(self, relation_id: int) -> str:
-        """Return the policy name for the given relation id."""
-        return f"charm-autounseal-{relation_id}"
+    def create_transit_key(self, mount_point: str, key_name: str) -> None:
+        """Create a new key in the transit backend."""
+        response = self._client.secrets.transit.create_key(mount_point=mount_point, name=key_name)
+        logging.debug(f"Created a new transit key. response={response}")
 
-    def _get_autounseal_approle_name(self, relation_id: int) -> str:
-        """Return the approle name for the given relation id."""
-        return f"charm-autounseal-{relation_id}"
+    def delete_role(self, role_name: str) -> None:
+        """Delete the approle with the given name."""
+        return self._client.auth.approle.delete_role(role_name)
 
-    def _get_autounseal_key_name(self, relation_id: int) -> str:
-        """Return the key name for the given relation id."""
-        return str(relation_id)
+    def delete_policy(self, policy_name: str) -> None:
+        """Delete the policy with the given name."""
+        return self._client.sys.delete_policy(policy_name)
 
-    def _create_autounseal_key(self, mount_point: str, relation_id: int) -> str:
-        """Create a new autounseal key."""
-        key_name = self._get_autounseal_key_name(relation_id)
-        response = self._client.secrets.transit.create_key(mount_point=mount_point, name=key_name)
+
+def generate_pem_bundle(certificate: str, private_key: str) -> str:
+    """Generate a PEM bundle from a certificate and private key."""
+    return f"{certificate}\n{private_key}"
+
+
+AUTOUNSEAL_POLICY = """path "{mount}/encrypt/{key_name}" {{
+    capabilities = ["update"]
+}}
+
+path "{mount}/decrypt/{key_name}" {{
+    capabilities = ["update"]
+}}
+"""
+
+
+@dataclass
+class VaultAutounsealRelationDetails:
```

Comment:
I'm somewhat confused by this file now, it used to be quite clear what it was (a Vault http api abstraction), now it's a stranger, larger thing that does a lot. 

Does this class really belong in `vault_client.py`?

Reply:
The same comment may apply to other classes here